### ISSUE TITLE
[FR] Add duration as a run attribute

### ISSUE THREAD

**Issue by wamartin-aml** [mlflow/mlflow#7682]
### Willingness to contribute

Yes. I can contribute this feature independently.

### Proposal Summary

For runs, add duration as a run attribute. It would only exist on runs that have both start_time and end_time set and be the simple difference of the two. Duration could then be used as either a filter or ordering parameter for search_runs.

### Motivation

> #### What is the use case for this feature?
It is valuable for users to be able to quickly pick out runs that are especially long or short.

> #### Why is this use case valuable to support for MLflow users in general?
Nobody wants their runs to take a long time if they don't have to, and finding short runs can also help isolate errors quickly.

> #### Why is this use case valuable to support for your project(s) or organization?
We have customers who want to be able to sort and filter by duration.

> #### Why is it currently difficult to achieve this use case?
Customers can use start_time and end_time in search_runs, but combining the two in a single query to even approximate duration is really messy.

### Details

Duration would just be updated on run info when end_time is updated. If start_time exists, duration would be set to the difference of the two.

### What component(s) does this bug affect?

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [X] `area/tracking`: Tracking Service, tracking client APIs, autologging

### What interface(s) does this bug affect?

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

### What language(s) does this bug affect?

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

### What integration(s) does this bug affect?

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

**Comment by dbczumar** [comment]
Thanks @wamartin-aml , can you put this proposal in a short document including the use cases that we can share with other stakeholders? Feel free to use this template: https://docs.google.com/document/d/1AQGgJk-hTkUo0lTkGqCGQOMelQmz05kQz_OA4bJWaJE/edit

**Comment by dbczumar** [comment]
^ Btw, the proposal sounds good to me.

**Comment by wamartin-aml** [comment]
@dbczumar here's [the doc](https://docs.google.com/document/d/1OFTMIb4c42XjENjhyPlX5tC22-6o_fguElX7NbowdAs/edit?usp=sharing) you requested, please feel free to ask for clarifications/expansions if anything is missing!

**Comment by dbczumar** [comment]
Thanks @wamartin-aml ! Since duration is derived from start time & end time, I think it might be nice to avoid putting that derived information in the RunInfo.

What are your thoughts about supporting duration in search as a derived attribute, e.g. SearchRuns with filter `attributes.duration < 1000` and surfacing it on the `RunInfo` entity in the MLflow Python / R / Java clients?

**Comment by wamartin-aml** [comment]
Yeah that's fine! I'll fit this in when my other workload is a bit lower
